gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,335 @@
+        },
+        updateProcInfo: {
+            funcName: "gpii.processes.updateProcInfo",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        initProcInfoNotRunning: {
+            funcName: "gpii.processes.initProcInfoNotRunning",
+            args: ["{arguments}.0"]   // command name (string)
+        },
+        initMonitor: {
+            funcName: "gpii.processes.initMonitor",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        monitorStateChange: {
+            funcName: "gpii.processes.monitorStateChange",
+            args: ["{that}", "{arguments}.0"] // monitor info structure
+        },
+        monitorRunStateChanged: {
+            funcName: "gpii.processes.monitorRunStateChanged",
+            args: ["{that}", "{arguments}.0"] // monitor info structure
+        },
+        trackRunState: {
+            funcName: "gpii.processes.trackRunState",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            // process info, handler
+        },
+        stopTrackingRunState: {
+            funcName: "gpii.processes.stopTrackingRunState",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            // handler, intervalID
+        },
+        // Tracking *any* state change.
+        trackState: {
+            funcName: "gpii.processes.trackState",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            // process info, handler
+        },
+        stopTrackingState: {
+            funcName: "gpii.processes.stopTrackingState",
+            args: ["{that}", "{arguments}.0", "{arguments}.1"]
+            // handler, intervalID
+        }
+    }
+});
+
+// Return an list of process information structures corresponding to the
+// names of each of the passed in commands.
+gpii.processes.findSolutionsByCommands = function (commandNames) {
+    if (!!commandNames) {
+        return fluid.accumulate(commandNames, function (aCommand, matches) {
+            var procInfos = gpii.processes.findProcessesByCommand(aCommand);
+            matches = matches.concat(procInfos);
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+// Return an list of process information structures corresponding to the
+// pids pass in.
+gpii.processes.findSolutionsByPids = function (pids) {
+    if (!!pids) {
+        return fluid.accumulate(pids, function (aPid, matches) {
+            var found = gpii.processes.findProcessByPid(aPid);
+            if (found !== null) {
+                matches.push(found);
+            }
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+// Return THE process info object that matches the given process id.
+// Note that it can return "null" meaning there is no such process.
+gpii.processes.findProcessByPid = function (pid, procArray) {
+    if (!procArray) {
+        procArray = gpii.processes["native"].getProcessList(pid);
```

Comment:
RIght, allowing a pid was a recent change, and I should have updated the documentation.

I'd update the comment, but I can't find it.  Where does it say that only a name is allowed?